spec/classes/init_spec.rb

```diff
@@ -91,20 +91,13 @@
   describe 'with default values for parameters on' do
     platforms.sort.each do |k, v|
       context "#{k}" do
-        if v[:osfamily] == 'Solaris'
-          let :facts do
-            { :osfamily      => v[:osfamily],
-              :kernelrelease => v[:release],
-            }
-          end
-        else
-          let :facts do
-            { :lsbdistcodename           => v[:lsbdistcodename],
-              :osfamily                  => v[:osfamily],
-              :operatingsystemrelease    => v[:release],
-              :operatingsystemmajrelease => v[:majrelease],
-            }
-          end
+        let :facts do
+          { :lsbdistcodename           => v[:lsbdistcodename],
+            :osfamily                  => v[:osfamily],
+            :kernelrelease             => v[:release],        # Solaris specific
+            :operatingsystemrelease    => v[:release],        # Linux specific
+            :operatingsystemmajrelease => v[:majrelease],
+          }
```

Comment:
this saves code and works well
see https://github.com/ghoneycutt/puppet-module-nscd/blob/v1.7.0/spec/classes/init_spec.rb#L347-L352